### PR TITLE
Fixed #26717 -- Added a stream_class attribute to Serializer.

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://code.djangoproject.com/ticket/26717

### CODE REVIEW

File: django/core/serializers/base.py [hunk 1]
```diff
@@ -59,14 +59,15 @@ class Serializer(object):
     # internal Django use.
     internal_use_only = False
     progress_class = ProgressBar
+    stream = six.StringIO
```
Comment (charettes) [inline django/core/serializers/base.py]: `stream_class`?

Comment (berkerpeksag) [inline django/core/serializers/base.py]: Yes, stream_class is much better. I've renamed stream to stream_class. Thanks!

File: tests/serializers/tests.py [hunk 2]
```diff
@@ -353,6 +353,34 @@ def test_serialize_proxy_model(self):
...
+
+        class Serializer(serializers.json.Serializer):
+            stream_class = File
+
+        score = Score(score=3.4)
+        score.save()
+
+        serializer = Serializer()
+        data = serializer.serialize(Score.objects.all())
+        self.assertIs(serializer.stream_class, File)
```
Comment (charettes) [inline tests/serializers/tests.py]: You might want to `assertIsInstance(serializer.stream, File)` to make sure the `stream_class` attribute was actually used.

File: tests/serializers/tests.py [hunk 3]
```diff
@@ -353,6 +353,34 @@ def test_serialize_proxy_model(self):
...
+                self.lines.append(line)
+
+            def getvalue(self):
+                return ''.join(self.lines)
+
+        class Serializer(serializers.json.Serializer):
+            stream_class = File
+
+        score = Score(score=3.4)
+        score.save()
```
Comment (charettes) [inline tests/serializers/tests.py]: You should be able to use `SimpleTestCase` (which prevents any queries) by setting an `id` manually on your score instance, avoiding the `save()` call and passing a singleton list containing `score` to `serializer.serialize()`:

```python
data = serializer.serialize([Score(id=1, score=3.4)])
```

With this approach you should be able to hardcode the `"pk": 1` below as well.

File: tests/serializers/tests.py (hunk 2 above)
Comment (berkerpeksag) [inline tests/serializers/tests.py]: Done.

File: tests/serializers/tests.py (hunk 3 above)
Comment (berkerpeksag) [inline tests/serializers/tests.py]: Good idea, done. Thanks.

**Comment by timgraham** [comment]
merged in 724dd2043ec849e605ba4f4cf1430458c2a8d90a, thanks!

**Pull request closed without merging**